universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java

```diff
@@ -191,4 +195,36 @@ public int hashCode() {
     }
     return pathsRemoved.build();
   }
+
+  public void setUserName(String displayName) {
+    String firstName;
+    String lastName = null;
+    String middleName = null;
+    List<String> listSplit = Splitter.on(' ').splitToList(displayName);
+    switch (listSplit.size()) {
+      case 2:
+        firstName = listSplit.get(0);
+        lastName = listSplit.get(1);
+        break;
+      case 3:
+        firstName = listSplit.get(0);
+        middleName = listSplit.get(1);
+        lastName = listSplit.get(2);
+        break;
+      case 1:
+        // fallthrough
+      default:
+        // Too many names - put them all in first name.
+        firstName = displayName;
```

Comment:
https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/ :) My manager sent me this once and now I'm nervous whenever I need to write name logic

question: Do we need to split by first/middle/last, or could we just set their entire user/display name as their username?

Reply:
Yeah, I thought of that article while writing this.  In this case I am not omitting or disqualifying any names.  I am just doing some heuristic guessing - I don't think anyone will be insulted to have their display name split into first / last, or first/middle/last, by spaces.  This is inaccurate for some people - me, for example.  My legal name is `Nathan Douglas McKinley Henderson` - this code will give up and shove that into "first name".  But users like me will have the opportunity to change it if it's wrong, and it'll be right most of the time, so I think it's worth doing.